file-formats/intm/type/zif_aff_intm_v1.intf.abap

```diff
@@ -0,0 +1,111 @@
+INTERFACE zif_aff_intm_v1
+  PUBLIC.
+
+  TYPES:
+    "! <p class="shorttext">General Information</p>
+    "! General information
+    BEGIN OF ty_general_information,
+      "! <p class="shorttext">Model Name</p>
+      "! Model name
+      model_name           TYPE c LENGTH 30,
+      "! <p class="shorttext">Model Type</p>
+      "! Model type
+      model_type           TYPE c LENGTH 256,
+      "! <p class="shorttext">Intelligent Scenario</p>
+      "! Intelligent scenario of the model
+      intelligent_scenario TYPE c LENGTH 20,
+      "! <p class="shorttext">Modelling Context ID</p>
+      "! Modelling context ID which is the parent of the model
+      modelling_context_id TYPE c LENGTH 36,
+      "! <p class="shorttext">Adapter ID</p>
+      "! Adapter ID
+      adapter_id           TYPE c LENGTH 32,
+      "! <p class="shorttext">Native Model Type</p>
+      "! Native model type
+      native_model_type    TYPE c LENGTH 256,
+      "! <p class="shorttext">Model Specification</p>
+      "! Model specification
+      model_specification  TYPE string,
+      "! <p class="shorttext">Model Definition</p>
+      "! Model definition
+      model_definition     TYPE string,
+      "! <p class="shorttext">State</p>
+      "! State of the model
+      state                TYPE c LENGTH 10,
+      "! <p class="shorttext">Signature</p>
+      "! Model signature
+      signature            TYPE string,
+      "! <p class="shorttext">Bindings</p>
+      "! Model bindings
+      bindings             TYPE string,
+      "! <p class="shorttext">Is Enhancement</p>
+      "! If true, the model is an enhancement, if false the model is a new model.
+      is_enhancement       TYPE abap_bool,
+      "! <p class="shorttext">Executable ID</p>
+      "! Executable ID
+      executable_id        TYPE c LENGTH 64,
+      "! <p class="shorttext">Large Language Model Name</p>
+      "! Large language model name
+      llm_name             TYPE string,
+      "! <p class="shorttext">Large Language Model Version</p>
+      "! Large language model version
+      llm_version          TYPE c LENGTH 30,
+    END OF ty_general_information,
+
+    "! <p class="shorttext">Segment Field</p>
+    "! Model segment field
+    BEGIN OF ty_segment_field,
+      "! <p class="shorttext">Field Name</p>
+      "! Table or view field name
+      field_name     TYPE c LENGTH 30,
+      "! <p class="shorttext">Field Position</p>
+      "! Field position
+      field_position TYPE i,
```

Comment:
Can be fixed next week: 

We can discuss whether you really need "field_position".  I also saw it in INTS. Maybe it is sufficient if you have the right order of fields in your segment_fields table.

Reply:
Segment fields table is for multi active model functionality. Customer can have multiple INTMs for one INTS and keep them active also. This functionality is currently live in our Fiori and it's future scope for ADT. This field_position is to identify the position of same fields being selected in multiple models and to validate the position of fields in hash ID calculation. If the position of field is changed at any point of time during our operations, then hash ID will be mismatch the combination. We wanted to fix this field positions of segment fields from the creation of model itself and so during the hash calculation we will strictly follow the field position. We have hard dependency with this field for already delivered scenarios and so we cannot avoid this field.

Reply:
OK, I think I understood. Since this is a global position (not locally in one INTM object, but across several objects), a value is needed 👍